server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -88,6 +103,43 @@ private void checkProjectKeyConsistency(ScannerReport.Metadata reportMetadata) {
     }
   }
 
+  private void checkOrganizationKeyConsistency(ScannerReport.Metadata reportMetadata, Organization organization) {
+    String organizationKey = reportMetadata.getOrganizationKey();
+    String resolveReportOrganizationKey = resolveReportOrganizationKey(organizationKey);
+    if (!resolveReportOrganizationKey.equals(organization.getKey())) {
+      if (reportBelongsToDefaultOrganization(organizationKey)) {
+        throw MessageException.of(format(
+          "Report does not specify an OrganizationKey but it has been submitted to another organization (%s) than the default one (%s)",
```

Comment:
In fact this error can only be a bug in scanner. So indeed message can be technically exact. It could also be a ISE, but that's minor.